src/secret.js

```diff
@@ -4,25 +4,43 @@ import inquirer from 'inquirer';
 import { setRepoSecret } from './init';
 import { ConfigApi } from './services/ConfigApi';
 import { auth } from './login';
+import { wait } from './wait';
 
 export const setSecretKey = async (options) => {
 	if (!options.context || !options.context.searchspring || !options.context.project || !options.context.project.path) {
 		console.log(chalk.red(`Error: No Snap project found in ${process.cwd()}.`));
 		exit(1);
 	}
 
-	const answers = await inquirer.prompt({
-		type: 'input',
-		name: 'secretKey',
-		message: 'Please enter the secretKey as found in the SMC console (32 characters):',
-		validate: (input) => {
-			return input && input.length > 0 && /^[0-9a-zA-Z]{32}$/.test(input);
+	const questions = [
+		{
+			type: 'input',
+			name: 'secretKey',
+			message: 'Please enter the secretKey as found in the SMC console (32 characters):',
+			validate: (input) => {
+				return input && input.length > 0 && /^[0-9a-zA-Z]{32}$/.test(input);
+			},
 		},
-	});
+	];
+
+	if (typeof options.context.searchspring.siteId === 'object') {
+		const choices = Object.keys(options.context.searchspring.siteId);
+		if (choices.length) {
+			questions.unshift({
+				type: 'list',
+				name: 'siteId',
+				message: 'Please select which siteId to add/update the secretKey for:',
```

Comment:
Seems a bit confusing I think to ask for a secretKey first...

Reply:
This was using `unshift` so it would ask for the siteId first. I refactored this to use `when` so that it's more clear